app/helpers/reactjs_helper.rb

```diff
@@ -11,7 +11,18 @@ def react_component(name, props = {})
     content_tag('foreman-react-component', '', :name => name, :data => { props: props })
   end
 
+  def webpack_root_url(plugin)
+    "/assets/#{plugin.name}"
+  end
+
+  def load_plugin_module_script(plugin, prefix = nil)
+    # This should match webpack.config.js entries for plugin's federated module entry point.
+    entry_point = [prefix, 'index'].compact.join('_')
+    "window.tfm.tools.loadPluginModule(url: ""#{webpack_root_url(plugin)}"", scope: ""#{plugin.name}"", module: ""./#{entry_point}"");"
```

Comment:
I'm confused by the double quotes. Does that work?

Reply:
I think so, although the system is not in running state at the moment.
The idea is to inject a script tag that will look like:
``` html
<script>
window.tfm.tools.loadPluginModule(url: "plugin1.js", scope: "plugin1", module: "entry_point.js");
window.tfm.tools.loadPluginModule(url: "plugin2.js", scope: "plugin2", module: "entry_point.js");
</script>
```
This will effectively load the plugin as a federated module. This is the heart of the dynamic federated modules trick.

Reply:
Right, but I see `""` which looks like confused me. Perhaps it's better to use single quotes where possible, or use heredoc. So, wouldn't this work:
```suggestion
    "window.tfm.tools.loadPluginModule(url: '#{webpack_root_url(plugin)}', scope: '#{plugin.name}', module: './#{entry_point}');"
```

Or
```suggestion
    <<~JS
    window.tfm.tools.loadPluginModule(url: "#{webpack_root_url(plugin)}", scope: "#{plugin.name}", module: "./#{entry_point}");
    JS
```